source/common/router/config_impl.cc

```diff
@@ -109,6 +134,10 @@ HashPolicyImpl::HashPolicyImpl(
     case envoy::api::v2::RouteAction::HashPolicy::kHeader:
       hash_impls_.emplace_back(new HeaderHashMethod(hash_policy.header().header_name()));
       break;
+    case envoy::api::v2::RouteAction::HashPolicy::kCookie:
+      hash_impls_.emplace_back(
+          new CookieHashMethod(hash_policy.cookie().name(), hash_policy.cookie().ttl().seconds()));
```

Comment:
It would probably be more correct to convert from `Duration` to milliseconds and then round or something, but it's probably fine, I don't think anyone is going to care about fractional seconds here.